test/test_prototype_datasets_builtin.py

```diff
@@ -25,6 +27,17 @@ def extract_datapipes(dp):
     return get_all_graph_pipes(traverse(dp, only_datapipe=True))
 
 
+def consume(iterator):
+    # Copied from the official itertools recipes: https://docs.python.org/3/library/itertools.html#itertools-recipes
+    deque(iterator, maxlen=0)
+
+
+def next_consume(iterator):
```

Comment:
This is useful since we often just want the first sample, but need to make sure to still consume to avoid dangling streams. `list(iterator)` would also do the trick, but keeps everything in memory for no reason.